src/inmanta/module.py

```diff
@@ -1997,7 +2007,7 @@ def __init__(
                 self.virtualenv = env.VirtualEnv(venv_path)
 
         self.loaded = False
-        self.modules: dict[str, Module] = {}
+        self.modules: dict[str, Module[ModuleMetadata]] = {}
```

Comment:
Don't these changes raise as many errors as they solve? `TModuleMetadata` is invariant, so I would expect this to raise errors elsewhere. Iirc that's the reason I've left this one alone up till now.

Reply:
It removed a huge amount of warnings from the output. I only see now that I should have used TModuleMetadata instead of  ModuleMetadata. However, it did reduce the amount of errors a lot

Reply:
Ok, then we can let it slide. I'm pretty sure it's not entirely correct, but perhaps it's a step in the right direction. To be honest, I'd really like it if `mypy` would no longer assume `Any` for missing generic parameters. Sometimes you just don't want to be as verbose.